docs/blog/2024-12-10-tailoring-nx-for-your-organization.md

```diff
@@ -0,0 +1,182 @@
+    name: projectName,
+    directory,
+    importPath,
+    tags: [`type:${options.type}`, `scope:${shared}`],
+    linter: Linter.EsLint,
+    style: 'css',
+    unitTestRunner: 'vitest',
+  });
+
+  return callbackAfterFilesUpdated;
+}
+
+export default libraryGenerator;
+```
+
+This example of a generator doesn’t do much but pass in options to an existing generator. However, it will consistently:
+
+- Name the project appropriately
+- Create the correct import path
+- Place new projects in the correct directory
+- Add the proper tags to the project
+- Select the correct set of tools
+
+All of that without having to refer to your documentation on how to run the generator the right way. And most of the functionality is provided by Nx! You just need to provide some logic specific to your organization.
+
+These generators will also show up in [Nx Console](/getting-started/editor-setup) alongside the other generators from Nx plugins, so engineers will have quick access.
+
+{% youtube src="https://www.youtube.com/watch?v=myqfGDWC2go" /%}
+
+## Configure your tools in one place with custom inferred tasks
+
+[Inferred tasks](/concepts/inferred-tasks) are a powerful option for reducing the configuration of tasks in your monorepo. Each plugin with inferred tasks offers some configuration, but what if you need to adjust something that’s not configurable out of the box? What if there’s a tool that we don’t have a plugin for?
+
+Writing your own task inference plugin allows you handle these situations. Let’s say the hottest, coolest new tool hits the ecosystem, and you need to use it today. You could drop an issue in GitHub asking for the core team to support it. Or you could write it yourself and use it today. Imagine this new tool is named `zebra` and has a config file `zebra.json` for each project. The inferred task function for this could look something like this:
+
+```typescript
+export const createNodes: CreateNodes = [
+  '**/zebra.json', // search workspace for files named zebra.json
+  (fileName: string, opts, context: CreateNodesContext) => {
+    // nodes on the project graph are set by a source root
+    const root = dirname(fileName);
+
+    // add a zebra task to that project
+    return {
+      projects: {
+        [root]: {
+          targets: {
+            zebra: {
+              command: `zebra .`,
+            },
+          },
+        },
+      },
+    };
+  },
+];
+```
+
+This is a very naive implementation, but it gives you an idea of what implementing these task inference plugins looks like. [Check out this much more developed example.](/extending-nx/tutorials/tooling-plugin)
+
+## Publish to your org with `nx release`
+
+Plugins can also be bundled and published for your organization. Since we’re the monorepo people, it might seem wild to suggest that you would have multiple monorepos in your organization; but this is a common scenario, and we whole-heartedly support it.
```

Comment:
again emphasize that you can just have plugins in your monorepo and run them right away without any further compilatio or anything. Just so we repeat this and ppl don't miss it.